tests/gradients/finite_diff/test_finite_difference.py

```diff
@@ -417,7 +417,7 @@ def test_independent_parameters(self):
             j1 = fn(dev.execute(tapes))
 
         # We should only be executing the device to differentiate 1 parameter (2 executions)
-        assert tracker.latest["executions"] == 2
+        assert tracker.totals["executions"] == 2
```

Comment:
Why aren't executions included in `tracker.latest` anymore? Does this count as a breaking change?

Reply:
`latest` is the last thing that was passed to `update`. 

Now the "batch" information is updated after the individual circuit information, so `"executions"` is no longer part of the last information to be updated.

`totals` is more relevant to this test anyway.

Reply:
I pretty much blindly changed everything from `dev.num_executions` to `tracker.latest["executions"]` while re-factoring tests, so agreed that this is a fine thing to fix